src/domain/services/product-validation/edit.ts

```diff
@@ -0,0 +1,24 @@
+import { NextFunction, Request, Response } from 'express';
+
+export default class ProductValidator {
+  static editProduct(req: Request, res: Response, next: NextFunction) {
+    const bodyRequest = req.body;
+    const productId = Number(req.params.id);
+
+    if (isNaN(productId)) {
+      return res.status(400).send({ message: 'product id must be a number' });
+    }
+
+    if (
+      bodyRequest.categoryId ||
+      bodyRequest.name ||
+      bodyRequest.description ||
+      bodyRequest.price ||
+      req.file
+    ) {
+      return next();
+    }
+
+    return res.status(400).send({ message: 'invalid body request' });
```

Comment:
Shouldn't this code containing the status code be in the controller?

Reply:
the controller sends to client the result of service. In the commented code section  the service returns message and status. I think this approach is clearer than the older. WDYT?